code/controllers/subsystem/fluids/fluid_effects.dm

```diff
@@ -0,0 +1,83 @@
+/**
+ *
+ */
+SUBSYSTEM_DEF(fluid_effect)
+	name = "Fluid Effect"
+	wait = 1
+	flags = SS_BACKGROUND|SS_POST_FIRE_TIMING
+
+	//
+	/// The amount of time
+	var/effect_wait = 1 SECONDS
+	///
+	var/list/list/obj/effect/particle_effect/fluid/processing
+	///
+	var/tmp/list/obj/effect/particle_effect/fluid/currentrun
+	///
+	var/tmp/num_buckets
+	///
+	var/tmp/processing_bucket_index
+
+/datum/controller/subsystem/fluid_effect/Initialize(start_timeofday)
+	if (effect_wait < wait)
+		if (effect_wait > 0)
+			wait = effect_wait
+		else
+			effect_wait = wait
```

Comment:
What's going on here exactly?

Reply:
If the effect wait is less than the fire wait it shifts the fire wait down to the effect wait so the SS can process the effects at that wait.

If the effect wait is 0 it just assumes that the intended effect wait is the same as the fire wait.

Reply:
Has changed a bit in the current implementation and it has comments.